Crash just after pressing synchronise option on android client

### Steps to reproduce
1. Don't know if reproducable
2. After syncing 3 or 4 directories (which probably hadn't finished) pressed sync on another directory and the client crashed immediately
3. 

### Expected behaviour
- Tell us what should happen
No crash
### Actual behaviour
- Tell us what happens
************ CAUSE OF ERROR ************

java.lang.RuntimeException: Unable to start service com.owncloud.android.services.OperationsService@663e908 with Intent { act=SYNC_FOLDER cmp=com.nextcloud.client/com.owncloud.android.services.OperationsService (has extras) }: java.lang.ClassCastException: com.nextcloud.client.account.RegisteredUser cannot be cast to android.accounts.Account
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4179)
	at android.app.ActivityThread.-wrap21(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2119)
	at android.os.Handler.dispatchMessage(Handler.java:108)
	at android.os.Looper.loop(Looper.java:166)
	at android.app.ActivityThread.main(ActivityThread.java:7529)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
Caused by: java.lang.ClassCastException: com.nextcloud.client.account.RegisteredUser cannot be cast to android.accounts.Account
	at com.owncloud.android.services.OperationsService.onStartCommand(OperationsService.java:196)
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4151)
	... 8 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30120051
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: HUAWEI
Device: HWAGS2
Model: AGS2-W09
Id: HUAWEIAGS2-W09
Product: AGS2-W09EEA

************ FIRMWARE ************
SDK: 26
Release: 8.0.0
Incremental: 278(OCEC431)

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

### Environment data
Android version:

Device model: 

Stock or customized system:

Nextcloud app version:

Nextcloud server version:

Reverse proxy:

### Logs
#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

--- a/src/main/java/com/nextcloud/client/jobs/ContactsBackupWork.kt
+++ b/src/main/java/com/nextcloud/client/jobs/ContactsBackupWork.kt
@@ -166,9 +166,9 @@ class ContactsBackupWork(
         )
     }
 
-    private fun expireFiles(daysToExpire: Int, backupFolderString: String, account: User) { // -1 disables expiration
+    private fun expireFiles(daysToExpire: Int, backupFolderString: String, user: User) { // -1 disables expiration
         if (daysToExpire > -1) {
-            val storageManager = FileDataStorageManager(account.toPlatformAccount(),
+            val storageManager = FileDataStorageManager(user.toPlatformAccount(),
                 applicationContext.getContentResolver())
             val backupFolder: OCFile = storageManager.getFileByPath(backupFolderString)
             val cal = Calendar.getInstance()
@@ -184,7 +184,7 @@ class ContactsBackupWork(
                     // delete backups
                     val service = Intent(applicationContext, OperationsService::class.java)
                     service.action = OperationsService.ACTION_REMOVE
-                    service.putExtra(OperationsService.EXTRA_ACCOUNT, account)
+                    service.putExtra(OperationsService.EXTRA_ACCOUNT, user.toPlatformAccount())
                     service.putExtra(OperationsService.EXTRA_REMOTE_PATH, backup.remotePath)
                     service.putExtra(OperationsService.EXTRA_REMOVE_ONLY_LOCAL, false)
                     operationsServiceBinder!!.queueNewOperation(service)

